lib/workers/pr/index.ts

```diff
@@ -21,14 +22,61 @@ function noLeadingAtSymbol(input: string): string {
   return input.length && input.startsWith('@') ? input.slice(1) : input;
 }
 
-// TODO: fix types
+function assigneesFromConfig(config: BranchConfig): Promise<string[]> {
+  return Promise.resolve(config.assignees || []);
+}
+
+async function assigneesFromCodeowners(
+  config: BranchConfig,
+  pr: Pr
+): Promise<string[]> {
+  if (!config.assigneesFromCodeowners) {
+    return [];
+  }
+
+  const codeowners =
+    (await platform.getFile('CODEOWNERS', pr.targetBranch)) ||
+    (await platform.getFile('.github/CODEOWNERS', pr.targetBranch)) ||
+    (await platform.getFile('.gitlab/CODEOWNERS', pr.targetBranch)) ||
+    (await platform.getFile('docs/CODEOWNERS', pr.targetBranch));
+
+  if (!codeowners) {
+    return [];
+  }
+
+  const prFiles = await platform.getPrFiles(pr.number);
+  const rules = codeowners
+    .split('\n')
+    .filter((line) => line && !line.startsWith('#'))
+    .map((line) => {
+      const [pattern, ...usernames] = line.split(/\s+/);
+      return {
+        usernames,
+        match: (path: string) => {
+          const matcher = ignore().add(pattern);
+          return matcher.ignores(path);
+        },
+      };
+    })
+    .reverse();
+
+  const matchingRule = rules.find((rule) => prFiles.every(rule.match));
+  if (!matchingRule) {
+    return [];
+  }
+  return matchingRule.usernames.map(noLeadingAtSymbol);
```

Comment:
Borrowed the idea here from https://github.com/beaugunderson/codeowners/blob/master/codeowners.js

Reply:
OK, `codeowners` license is MIT